### PR TITLE
test/pybind/test_rados.py: test test_operate_aio_write_op()

### PR DESCRIPTION
**Pull request by zjcmszh**
Signed-off-by: Zhang Jiao <zhangjiao@cmss.chinamobile.com>

### CODE REVIEW

**Comment by zjcmszh** [comment]
@tchaikov   Please check. 
 Thanks

File: src/test/pybind/test_rados.py [hunk 1]
```diff
@@ -598,6 +598,26 @@ def test_locator(self):
...
+        def cb(blah):
+            with lock:
+                count[0] += 1
+                lock.notify()
+            return 0
+        write_op = self.ioctx.create_write_op()
+        write_op.write(b'rzx')
+        comp = self.ioctx.operate_aio_write_op(write_op, "object", cb, cb)
+        comp.wait_for_complete()
+        comp.wait_for_safe()
```
Comment (tchaikov) [inline src/test/pybind/test_rados.py]: i don't think `wait_for_safe()` is necessary.

Comment (zjcmszh) [reply src/test/pybind/test_rados.py]: @tchaikov Updated. 
Thanks a lot.

**Comment by zjcmszh** [comment]
@batrick please review. Thanks

File: src/test/pybind/test_rados.py [hunk 2]
```diff
@@ -598,6 +598,25 @@ def test_locator(self):
...
 
+    def test_operate_aio_write_op(self):
+        lock = threading.Condition()
+        count = [0]
+        def cb(blah):
+            with lock:
+                count[0] += 1
+                lock.notify()
+            return 0
+        write_op = self.ioctx.create_write_op()
```
Comment (tchaikov) [inline src/test/pybind/test_rados.py]: could you use `WriteOpCtx` instead?

Comment (zjcmszh) [reply src/test/pybind/test_rados.py]: @tchaikov   Yes, i have tested it.

**Comment by zy751713126** [comment]
jenkins test make check

**Comment by zjcmszh** [comment]
@tchaikov 
Test Code:
```
from rados import Rados
from nose.tools import eq_ as eq, ok_ as ok, assert_raises
import threading
import rados
import unittest

class TestIoctx(unittest.TestCase):
    def setUp(self):
        self.rados = Rados(conffile='/var/lib/ceph/etc/ceph/ceph.conf')
        self.rados.connect()
        self.rados.create_pool('test_pool')
        assert self.rados.pool_exists('test_pool')
        self.ioctx = self.rados.open_ioctx('test_pool')
    def tearDown(self):
        self.ioctx.close()
        self.rados.delete_pool('test_pool')
        self.rados.shutdown()
    def test_operate_aio_write_op(self):
        lock = threading.Condition()
        count = [0]
        def cb(blah):
            with lock:
                count[0] += 1
                lock.notify()
            return 0
        write_op = self.ioctx.create_write_op()
        write_op.write(b'rzx')
        comp = self.ioctx.operate_aio_write_op(write_op, "object", cb, cb)
        comp.wait_for_complete()
        with lock:
            while count[0] < 2:
                lock.wait()
        eq(comp.get_return_value(), 0)
        eq(self.ioctx.read('object'), b'rzx')
        self.ioctx.release_write_op(write_op)

if __name__ == '__main__':
    unittest.main()
```

Test Results :
```
[root@jiao build]# python test_operate_aio_write_op.py 
.
----------------------------------------------------------------------
Ran 1 test in 2.563s

OK
[root@jiao build]# 
```

**Comment by zjcmszh** [comment]
@tchaikov  would you mind take a review? thanks

File: src/test/pybind/test_rados.py (hunk 2 above)
Comment (zjcmszh) [reply src/test/pybind/test_rados.py]: @ tchaikov  I already deferred to your suggestion to revise the code, please re-confirm.

**Comment by tchaikov** [comment]
- http://pulpito.ceph.com/kchai-2019-11-13_13:53:42-rados-wip-kefu-testing-2019-11-13-1728-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-11-15_11:39:33-rados-wip-kefu-testing-2019-11-13-1728-distro-basic-smithi/
- http://pulpito.ceph.com/kchai-2019-11-15_17:09:33-rados-wip-kefu-testing-2019-11-13-1728-distro-basic-mira/
- http://pulpito.ceph.com/kchai-2019-11-16_10:02:59-rados-wip-kefu-testing-2019-11-16-1438-distro-basic-mira/

---

- the test.sh failures are fixed by #31689
- test_cls_rbd.sh already tracked by https://tracker.ceph.com/issues/42333, and it's fixed in latest master.
- test_diskprediction_local tracked by https://tracker.ceph.com/issues/42764

**Pull request merged**